rust/private/rustc.bzl

```diff
@@ -675,7 +675,7 @@ def construct_arguments(
     ))
 
     # Set the SYSROOT to the directory of the rust_std files passed to the toolchain
-    env["SYSROOT"] = paths.dirname(toolchain.rust_std.files.to_list()[0].short_path)
+    env["SYSROOT"] = find_sysroot(toolchain)
```

Comment:
I'd prefer if this change was in a separate PR, the logic in `find_sysroot` is non-trivial and it differs from what was here before. I'd like to separate changes in behavior from refactorings. And looking into the implementation of `find_sysroot` I seem to have more questions, and I'd value discussing them in a focused change.